autonomi/examples/put_and_dir_upload.rs

```diff
@@ -0,0 +1,24 @@
+use autonomi::{Bytes, Client, EvmNetwork, Wallet};
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Default wallet of testnet.
+    let key = "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80";
+
+    let client = Client::connect(&["/ip4/127.0.0.1/udp/1234/quic-v1".parse()?]).await?;
+    let wallet = Wallet::new_from_private_key(EvmNetwork::ArbitrumSepolia, key)?;
+
+    // Put and fetch data.
+    let data_addr = client
+        .data_put(Bytes::from("Hello, World"), wallet.clone().into())
```

Comment:
```suggestion
        .data_put(Bytes::from("Hello, World"), &wallet.into())
```

Reply:
This yields

```
mismatched types
  expected enum `autonomi::client::payment::PaymentOption`
found reference `&_`
```

Reply:
Oh sorry!

Should be:

```rust
        .data_put(Bytes::from("Hello, World"), (&wallet).into())
```

But like I said in my other comment, this is the same as cloning under the hood in the `From` implementation, so I suppose it doesn't matter too much.